virttest/utils_misc.py

```diff
@@ -3358,7 +3360,7 @@ def verify_host_dmesg(dmesg_log_file=None, trace_re=None):
                 err += " Please check host dmesg log in debug log."
                 logging.debug(d_log)
             process.system("dmesg -C", ignore_status=True)
-            raise exceptions.TestError(err)
+            raise exceptions.TestFail(err)
```

Comment:
This introduces a change of behavior (a good one, I'd argue): From an Error, which is considered a configuration error (AKA not a bug in the software tested) to a Failure. Since KVM, and other virt backends have a kernel component, a kernel panic is likely to be a test failure, not an error.

So I agree with this change, and giving the tests the ability to add additional error patterns seems like a good addition.